clippy_lints/src/no_effect_replace.rs

```diff
@@ -0,0 +1,57 @@
+use clippy_utils::diagnostics::span_lint;
+use if_chain::if_chain;
+use rustc_ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `replace` statements which have no effect.
+    ///
+    /// ### Why is this bad?
+    /// It's either a mistake or confusing.
+    ///
+    /// ### Example
+    /// ```rust
+    /// "str".replace("123", "123");
+    /// "str".replacen("123", "123", 1);
```

Comment:
I just noticed something real quick :)

This code block is unclosed:
```suggestion
    /// "str".replacen("123", "123", 1);
    ```
```

It would probably also be better to make the examples more in line with the tests (`"12345".replace("123", "123"`).

Reply:
Oh many thanks :)